### PR TITLE
New completion functions

### PR DESCRIPTION
**Pull request by gabbaxx**
Added completion functions for all engine commands and the app flag

### CODE REVIEW

File: pkg/boot/flags.go [hunk 1]
```diff
@@ -54,8 +55,49 @@ func InjectAppWebSocketFlags(command *cobra.Command, hideEngineSpecificAppComple
...
+	} else {
+		ctx, global, _ := comm.OpenGlobalSocket()
+		docList, err := global.GetDocList(ctx)
+
+		if err != nil {
+			return []string{}, cobra.ShellCompDirectiveError
+		}
+		completions := make([]string, 0)
+		for _, x := range docList {
+			if strings.HasPrefix(x.DocName, toComplete) {
```
Comment (glooms) [inline pkg/boot/flags.go:95]: You can filter the results based on what to complete but the completion also does this, so it's not necessary, but it doesn't hurt either.

File: pkg/boot/flags.go [hunk 2]
```diff
@@ -54,8 +55,49 @@ func InjectAppWebSocketFlags(command *cobra.Command, hideEngineSpecificAppComple
...
+				completions = append(completions, x.DocName)
+			} else if strings.HasPrefix(x.DocId, toComplete) {
+				completions = append(completions, x.DocId)
+			}
+		}
+		return completions, 0
```
Comment (glooms) [inline pkg/boot/flags.go:101]: Should the `ShellCompDirective` be `cobra.ShellCompDirectiveNoFileComp` here as well?

File: pkg/commands/engine/analyze.go [hunk 3]
```diff
@@ -117,10 +118,11 @@ func CreateGetKeysCommand() *cobra.Command {
 
 func CreateEvalCommand() *cobra.Command {
 	return &cobra.Command{
-		Use:   "eval <measure 1> [<measure 2...>] by <dimension 1> [<dimension 2...]",
-		Args:  cobra.MinimumNArgs(1),
-		Short: "Evaluate a list of measures and dimensions",
-		Long:  `Evaluate a list of measures and dimensions. To evaluate a measure for a specific dimension use the <measure> by <dimension> notation. If dimensions are omitted then the eval will be evaluated over all dimensions.`,
+		Use:               "eval <measure 1> [<measure 2...>] by <dimension 1> [<dimension 2...]",
+		Args:              cobra.MinimumNArgs(1),
+		ValidArgsFunction: listFieldsForCompletion,
```
Comment (glooms) [inline pkg/commands/engine/analyze.go:123]: I don't think `eval` should have this completion.
If `eval` should have completion, it has to be tailored to this command which is quite different from others.

Comment (gabbaxx) [reply pkg/commands/engine/analyze.go:123]: Yep lets keep that one for later

**Pull request merged**